openslides_backend/action/mediafile/create_directory.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict
+
+from ...models.models import Mediafile
+from ...shared.patterns import Collection, FullQualifiedId
+from ..default_schema import DefaultSchema
+from ..generics import CreateAction
+from ..register import register_action
+
+
+@register_action("mediafile.create_directory")
+class MediafileUpdate(CreateAction):
+    """
+    Action to create directory a mediafile.
+    """
+
+    model = Mediafile()
+    schema = DefaultSchema(Mediafile()).get_create_schema(
+        required_properties=["meeting_id", "title"],
+        optional_properties=["access_group_ids", "parent_id"],
+    )
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Calculate inherited_access_group_ids and inherited_access_group_ids, if
+        access_group_ids are given.
+        """
+        instance["is_directory"] = True
+        if instance.get("parent_id") is not None:
+            parent = self.database.get(
+                FullQualifiedId(Collection("mediafile"), instance["parent_id"]),
+                ["inherited_access_group_ids"],
+            )
+            if not parent.get("inherited_access_group_ids"):
+                instance["inherited_access_group_ids"] = instance.get(
+                    "access_group_ids", []
+                )
+            elif not instance.get("access_group_ids"):
+                instance["inherited_access_group_ids"] = parent[
+                    "inherited_access_group_ids"
+                ]
+            else:
+                instance["inherited_access_group_ids"] = [
+                    id_
+                    for id_ in instance["access_group_ids"]
+                    if id_ in parent["inherited_access_group_ids"]
+                ]
+            instance["has_inherited_access_groups"] = (
+                len(instance["inherited_access_group_ids"]) > 0
+                or len(parent["inherited_access_group_ids"]) > 0
+            )
+        else:
+            instance["inherited_access_group_ids"] = instance.get(
+                "access_group_ids", []
+            )
+            instance["has_inherited_access_groups"] = (
+                len(instance["inherited_access_group_ids"]) > 0
+            )
```

Comment:
Since this should be the same as for `mediafile.update` (except the part about the children I commented there), you should move this to a seperate method/base class. You can wait until https://github.com/OpenSlides/openslides-backend/pull/216 is merged with that

Reply:
Done, with the code from mediafile.update.